packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,84 @@
-use std::path::PathBuf;
-
 use memory::shared_memory::MemoryId;
 
 use crate::{
-    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result},
+    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result, Error},
     proto::ExperimentId,
 };
 
+// TODO: move this to top-level
+#[derive(PartialEq, Eq, Clone)]
+pub enum EngineExitStatus {
+    Success,
+    Error,
+}
+
 // TODO: move this to top-level
 /// Shared memory cleanup in the process hard crash case.
 /// Not required for pod instances.
-pub fn cleanup_experiment(experiment_id: &ExperimentId) -> Result<()> {
+pub fn cleanup_experiment(
+    experiment_id: &ExperimentId,
+    exit_status: EngineExitStatus,
+) -> Result<()> {
     MemoryId::clean_up(experiment_id)?;
 
-    // TODO: We don't want to be deleting the parts files by default. We should figure out what to
-    //   do with this.
-    // remove_experiment_parts(experiment_id).or_else(|e| {
-    //     let err_string = e.to_string();
-    //     if err_string.contains("kind: NotFound") {
-    //         Ok(())
-    //     } else {
-    //         Err(Error::from(format!(
-    //             "Error while trying to remove experiment parts folder: {err_string}"
-    //         )))
-    //     }
-    // })?;
+    // Cleanup python socket files in case the engine didn't
+    let frompy_files = glob::glob(&format!("{experiment_id}-frompy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+    let topy_files = glob::glob(&format!("{experiment_id}-topy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+
+    frompy_files
+        .chain(topy_files)
+        .filter_map(std::result::Result::ok)
+        .for_each(|path| match std::fs::remove_file(&path) {
+            Ok(_) => {
+                match exit_status {
+                    EngineExitStatus::Success => tracing::error!(
+                        experiment = %experiment_id,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                    EngineExitStatus::Error => tracing::warn!(
+                        experiment = %experiment_id,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                };
+            }
+            Err(err) => {
+                tracing::warn!(
+                    experiment = %experiment_id,
+                    "Could not clean up {path:?}: {err}"
+                );
+            }
+        });
+
+    remove_experiment_parts(experiment_id)?;
 
     Ok(())
 }
 
-#[allow(dead_code)]
 fn remove_experiment_parts(experiment_id: &ExperimentId) -> Result<()> {
```

Comment:
I don't think we return a `Result` (or I missed a `?`)? Let's remove it then.

Reply:
Yeah no more return https://github.com/hashintel/hash/pull/477/commits/da069d85109c5b1bd00adc682770a33d8ae5a3e1